cobbler/settings/migrations/V3_4_0.py

```diff
@@ -269,8 +187,22 @@ def migrate(settings: dict) -> dict:
     if not V3_3_2.validate(settings):
         raise SchemaError("V3.3.2: Schema error while validating")
 
-    # rename keys and update their value
-    # add missing keys
-    # name - value pairs
+    # rename keys and update their value if needed
+    include = settings.pop("include")
+
+    # Drop defaults
+    from cobbler.settings import Settings
+
+    helper.key_drop_if_default(settings, Settings().to_dict())
+
+    # Write settings to disk
+    from cobbler.settings import update_settings_file
```

Comment:
I'm a bit surprised `pylint` isn't complaining about placing the imports inside `migrate()`. Is there a reason we don't have them grouped at the top of the file?

Reply:
Yes because otherwise we have a circular import.

Reply:
That is the thing I spoke about a few days ago. If we place them at the top of the file we get cyclical imports.

Reply:
Right, you mentioned that. Can you help me remember what import cycle is?

Reply:
Here we import the cobbler settings and in the Settings init.py we import the migrations, so we get an infinite loop of imports

Reply:
Thanks for the clarification!